esi/src/document.rs

```diff
@@ -0,0 +1,27 @@
+use crate::Result;
+use fastly::{http::request::PendingRequest, Request};
+
+/// A section of the pending response, either raw XML data or a pending fragment request.
+pub enum Element {
+    Raw(Vec<u8>),
+    Fragment(
+        // Metadata of the request
+        Request,
```

Comment:
Based on the documentation, I think you shouldn't need to store the original `Request`. It looks like you can call [PendingRequest::sent_req](https://docs.rs/fastly/latest/fastly/http/request/struct.PendingRequest.html#method.sent_req) to get a reference to it later.